ffi/src/lib.rs

```diff
@@ -41,13 +42,8 @@ pub unsafe trait Handle {
 /// not to dereference pointers whose referents don't implement `ReprC`; they are considered opaque
 pub unsafe trait ReprC: Sized {}
 
-/// Used to do a cheap reference-to-[`ReprC`]-reference conversion
-pub trait AsReprCRef<'itm> {
-    /// Robust C ABI compliant representation of &[`Self`]
-    type Target: ReprC + 'itm;
-
-    /// Convert from &[`Self`] into [`Self::Target`].
-    fn as_ref(&'itm self) -> Self::Target;
+pub trait FfiType {
```

Comment:
I think if possible we should do:

```rust
trait IntoFfi: FfiType {
    type Target: <Self as FfiType>::ReprC
}

trait TryFromReprC: FfiType {
    type Source: <Self as FfiType>::ReprC
}
```

Reply:
this is not valid because `<Self as FfiType>::ReprC` is a type and not a trait that you can use as a bound to `type Target`

Reply:
right, maybe then use this type in fn signature of `into_ffi` and `try_from_repr_c`?

Reply:
check it out now